cmd/kata-local-ci

```diff
@@ -0,0 +1,260 @@
+## Options include:
+##   -h: Print help
+##   -d <distro>: Select VM distro (fedora,ubuntu,centos, default:fedora)
+##   -j <job>: CI job to run (default: all)
+##   -w <dir>: Work directory to test (default $GOPATH/src/github.com/kata-containers)
+##   -f: force install of the VM (and overwrite disk)
+##   -F: "Faster" by reusing the image without reverting to earlier snapshot
+##
+## Creation options include:
+##   -k <ssh_key_file> (default ~/.ssh/id_rsa.pub)
+##   -o <disk-image>: Path and image name for the VM
+##   -s <size>: Disk image size for the VM (default 50G)
+##   -m <memory>: Memory for the VM in megabytes (default 8192M)
+##   -c <cpus>: Number of CPUs for the VM (default 4)
+##   -n network config (optional)
+##
+
+distro=fedora
+size=50G
+memory=8192
+cpus=4
+key=~/.ssh/id_rsa.pub
+vmdir=go/src/github.com/kata-containers
+directory=${GOPATH:-~go}/src/github.com/kata-containers
+disk=""
+network=""
+job=""
+kci=~/.kata-containers-ci
+install=false
+revert=true
+
+function help {
+    sed -E '/^##/s/^##\s?//g;t;d' $0
+}
+
+function progress() {
+    printf "******************************************** [$(date +%H:%M:%S)]\n"
+    printf "*\n"
+    printf "* $1\n"
+    printf "*\n"
+    printf "********************************************\n"
+}
+
+function sys() {
+    echo "# sudo" "$@"
+    sudo "$@"
+}
+
+function run() {
+    echo "%" "$@"
+    "$@"
+}
+
+function ci_cmd() {
+    echo "$ ssh kata@$VM_IPADDR -o StrictHostKeyChecking=no -o UpdateHostKeys=no" "\"$@\""
+    ssh kata@$VM_IPADDR -o StrictHostKeyChecking=no -o UpdateHostKeys=no "$@"
+}
+
+function ci_root_cmd() {
+    echo "$ ssh root@$VM_IPADDR -o StrictHostKeyChecking=no -o UpdateHostKeys=no" "\"$@\""
+    ssh root@$VM_IPADDR -o StrictHostKeyChecking=no -o UpdateHostKeys=no "$@"
+}
+
+while getopts fFhk:d:o:s:n:w:j: flag
+do
+    case "${flag}" in
+        h) help; exit 0 ;;
+        d) distro=$OPTARG;;
+        w) directory=$OPTARG;;
+        j) job=$OPTARG;;
+        v) name=$OPTARG;;
+        k) key=$OPTARG;;
+	o) disk=$OPTARG;;
+	s) size=$OPTARG;;
+	c) cpus=$OPTARG;;
+	m) memory=$OPTARG;;
+	n) network=$OPTARG;;
+	f) install=true ;;
+        F) revert=false;;
+        *) help; exit 1;;
+    esac
+done
+
+if [ ! -d "$kci" ]; then
+    mkdir "$kci"
+fi
+
+if [ ! -f "$kci/"$distro-config ]; then
+    install=true
+fi
+
+if [ -z "$disk" ]; then
+    disk=/var/lib/libvirt/images/kata-$distro-ci.qcow2
+fi
+
+if [ ! -f "$disk" ] ; then
+    install=true
+fi
+
+if [ ! -d "$directory"/kata-containers ]; then
+    echo "ERROR: The work directory $directory does not contain a 'kata-containers' subdirectory"
+    exit 4
+fi
+if [ ! -d "$directory"/tests ]; then
+    echo "ERROR: The work directory $directory does not contain a 'tests' subdirectory"
+    exit 4
+fi
+
+
+function vm_install() {
+    distro_commands=""
+    firstboot_cmds=(
+	--firstboot-command 'useradd -m  kata -s /bin/bash'
+	--firstboot-command 'echo "kata ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/kata'
```

Comment:
- Should this be `All=(ALL:ALL)`?
- Is `/etc/sudoers.d/` guaranteed to exist for all distros?

Reply:
For the first one, I am not sure what this would buy us (I don't think we ever need to run anything as a non-root group). What did you have in mind?

For the second one, it exists on all the distros supported by the script.